File: LeetCode/Easy/2965-find-missing-and-repeated-values/2965-find-missing-and-repeated-values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        n = len(grid)
        N = n*n
        expected_sum = N*(N+1)/2
        actual_sum = 0
        squaredSumExp = N*(N+1)*(2*N + 1)/6
        squaredSumAct = 0
        for i in range(n):
            for j in range(n):
                actual_sum += grid[i][j]
                squaredSumAct += grid[i][j]*grid[i][j]
        diff = actual_sum - expected_sum
        Squared_diff = squaredSumAct - squaredSumExp
        sumXY = Squared_diff/diff
        missing = int((sumXY - diff)/2)
        repeated = int((sumXY + diff)/2)
        return [repeated,missing]
        
        
        
        
        

File: LeetCode/Easy/2965-find-missing-and-repeated-values/2965-find-missing-and-repeated-values.cpp
class Solution {
public:
    vector<int> findMissingAndRepeatedValues(vector<vector<int>>& grid) {
        unordered_map<int,int> count;
        int n = grid.size();
        int summ = (n*n)*((n*n) + 1)/2;
        for(int i = 0 ; i < n ; i++){
            for(int j = 0 ; j < n ; j++){
                count[grid[i][j]]++;
            }
        }
        int s2 = 0;
        int r = 0;
        for(const auto& [k,v] : count){
                s2 += k;
                if( v == 2){
                    r = k;
                }
            
        }
        return {r,summ - s2};
    }
};